js/CustomerDatabase.js

```
@@ -1088,9 +1124,155 @@ class CustomerDatabase {
+        }
+
+        // Some long-inactive customers are permanently removed to prevent DB bloat
+        if (this.customers.size > 200) {
+            const toRemove = [];
+            this.customers.forEach((customer, id) => {
+                if (!customer.isActive && (this.engine.day - customer.lastVisit) > 60) {
+                    toRemove.push(id);
+                }
+            });
+            // Remove up to 10 stale entries per day
+            toRemove.slice(0, 10).forEach(id => this.customers.delete(id));
```

Biome (2.4.4)

[error] This callback passed to forEach() iterable method should not return a value.

(lint/suspicious/useIterableCallbackReturn)